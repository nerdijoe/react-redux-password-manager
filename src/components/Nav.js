import React from 'react'
import {Link} from 'react-router-dom'


export default class Nav extends React.Component {
  render() {
    return(
      <div>
        <Link to="/" >Home</Link><br/>
        <Link to="/list" >List</Link>
      </div>
    )
  }
}
