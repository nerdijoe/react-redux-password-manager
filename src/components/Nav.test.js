import React from 'react'
// import { expect } from 'chai';
import { shallow, mount } from 'enzyme';
import {Link} from 'react-router-dom'
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';

import Nav from './Nav'
import Style from './Nav.style'

describe('<Nav />', () => {

  it('should render Nav', () => {
    const wrapper = shallow(<Nav />)

    // expect(wrapper.containsAllMatchingElements([
    //   <AppBar/>,
    //   <Drawer>,
    //     <Link ><MenuItem>Home</MenuItem></Link>,
    //     <Link ><MenuItem>List</MenuItem></Link>,
    //   </Drawer>
    // ])).to.be.true

    // console.log(wrapper.instance());



    // expect(wrapper.contains(
    //   <div>
    //
    //     <AppBar
    //     />
    //
    //     <Drawer
    //     >
    //       <Link to="/" style={Style.link}><MenuItem>Home</MenuItem></Link>
    //       <Link to="/list" style={Style.link}><MenuItem>List</MenuItem></Link>
    //     </Drawer>
    //
    //   </div>
    // )).toBe(true)

    expect(wrapper).toMatchSnapshot();

  })


})

describe( '<Nav /> state', () => {
  it('should have state open:false in the beginning', () => {
    const wrapper = shallow(<Nav />)
    expect(wrapper.state('open')).toBe(false)
  })
})
