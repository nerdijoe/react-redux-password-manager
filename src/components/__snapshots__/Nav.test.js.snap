// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render Nav 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <AppBar
        onLeftIconButtonTouchTap={[Function]}
        showMenuIconButton={true}
        style={
            Object {
                "textAlign": "center",
              }
        }
        title="Password Manager"
        zDepth={1}
    />
    <Drawer
        disableSwipeToOpen={false}
        docked={false}
        onRequestChange={[Function]}
        open={false}
        openSecondary={false}
        swipeAreaWidth={30}
        width={200}
        zDepth={2}
    >
        <Link
            replace={false}
            style={
                Object {
                    "textDecoration": "none",
                  }
            }
            to="/"
        >
            <MenuItem
                anchorOrigin={
                    Object {
                        "horizontal": "right",
                        "vertical": "top",
                      }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                onTouchTap={[Function]}
                targetOrigin={
                    Object {
                        "horizontal": "left",
                        "vertical": "top",
                      }
                }
            >
                Home
            </MenuItem>
        </Link>
        <Link
            replace={false}
            style={
                Object {
                    "textDecoration": "none",
                  }
            }
            to="/list"
        >
            <MenuItem
                anchorOrigin={
                    Object {
                        "horizontal": "right",
                        "vertical": "top",
                      }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                onTouchTap={[Function]}
                targetOrigin={
                    Object {
                        "horizontal": "left",
                        "vertical": "top",
                      }
                }
            >
                List
            </MenuItem>
        </Link>
    </Drawer>
</div>,
  "nodes": Array [
    <div>
      <AppBar
            onLeftIconButtonTouchTap={[Function]}
            showMenuIconButton={true}
            style={
                  Object {
                        "textAlign": "center",
                      }
            }
            title="Password Manager"
            zDepth={1}
      />
      <Drawer
            disableSwipeToOpen={false}
            docked={false}
            onRequestChange={[Function]}
            open={false}
            openSecondary={false}
            swipeAreaWidth={30}
            width={200}
            zDepth={2}
      >
            <Link
                  replace={false}
                  style={
                        Object {
                              "textDecoration": "none",
                            }
                  }
                  to="/"
            >
                  <MenuItem
                        anchorOrigin={
                              Object {
                                    "horizontal": "right",
                                    "vertical": "top",
                                  }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        onTouchTap={[Function]}
                        targetOrigin={
                              Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
                        }
                  >
                        Home
                  </MenuItem>
            </Link>
            <Link
                  replace={false}
                  style={
                        Object {
                              "textDecoration": "none",
                            }
                  }
                  to="/list"
            >
                  <MenuItem
                        anchorOrigin={
                              Object {
                                    "horizontal": "right",
                                    "vertical": "top",
                                  }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        onTouchTap={[Function]}
                        targetOrigin={
                              Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
                        }
                  >
                        List
                  </MenuItem>
            </Link>
      </Drawer>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Nav />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Nav {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleClose": [Function],
        "handleToggle": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "open": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <AppBar
                    onLeftIconButtonTouchTap={[Function]}
                    showMenuIconButton={true}
                    style={
                              Object {
                                        "textAlign": "center",
                                      }
                    }
                    title="Password Manager"
                    zDepth={1}
          />
          <Drawer
                    disableSwipeToOpen={false}
                    docked={false}
                    onRequestChange={[Function]}
                    open={false}
                    openSecondary={false}
                    swipeAreaWidth={30}
                    width={200}
                    zDepth={2}
          >
                    <Link
                              replace={false}
                              style={
                                        Object {
                                                  "textDecoration": "none",
                                                }
                              }
                              to="/"
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        onTouchTap={[Function]}
                                        targetOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                              >
                                        Home
                              </MenuItem>
                    </Link>
                    <Link
                              replace={false}
                              style={
                                        Object {
                                                  "textDecoration": "none",
                                                }
                              }
                              to="/list"
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        onTouchTap={[Function]}
                                        targetOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                              >
                                        List
                              </MenuItem>
                    </Link>
          </Drawer>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <AppBar
                    onLeftIconButtonTouchTap={[Function]}
                    showMenuIconButton={true}
                    style={
                              Object {
                                        "textAlign": "center",
                                      }
                    }
                    title="Password Manager"
                    zDepth={1}
          />
          <Drawer
                    disableSwipeToOpen={false}
                    docked={false}
                    onRequestChange={[Function]}
                    open={false}
                    openSecondary={false}
                    swipeAreaWidth={30}
                    width={200}
                    zDepth={2}
          >
                    <Link
                              replace={false}
                              style={
                                        Object {
                                                  "textDecoration": "none",
                                                }
                              }
                              to="/"
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        onTouchTap={[Function]}
                                        targetOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                              >
                                        Home
                              </MenuItem>
                    </Link>
                    <Link
                              replace={false}
                              style={
                                        Object {
                                                  "textDecoration": "none",
                                                }
                              }
                              to="/list"
                    >
                              <MenuItem
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "right",
                                                            "vertical": "top",
                                                          }
                                        }
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        onTouchTap={[Function]}
                                        targetOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                              >
                                        List
                              </MenuItem>
                    </Link>
          </Drawer>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Nav />,
}
`;
